### ISSUE TITLE
Version 2015.04.26 doesn't like some output file names

### ISSUE THREAD

**Issue by RingoTheDog** [rg3/youtube-dl#5535]
The addedd FFMPEG post-processing in the latest version appears to choke on some file names.
(file names that don't have a specific extension I believe?...not sure)

C:\>youtube-dl -v "https://www.youtube.com/watch?v=IwfUnkBfdZ4" -o "C:\BATMAN.SU
PERMAN"
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'-v', u'https://www.youtube.com/watch?v=IwfUnkBfdZ4
', u'-o', u'C:\\BATMAN.SUPERMAN']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2015.04.26
[debug] Python version 2.7.8 - Windows-7-6.1.7601-SP1
[debug] exe versions: ffmpeg N-71346-gdf4fca2
[debug] Proxy map: {}
[youtube] IwfUnkBfdZ4: Downloading webpage
[youtube] IwfUnkBfdZ4: Extracting video information
[youtube] IwfUnkBfdZ4: Downloading DASH manifest
[debug] Invoking downloader on 'https://r8---sn-ni5f-t8gl.googlevideo.com/videop
layback?id=2307d49e405f759e&itag=137&source=youtube&requiressl=yes&ms=au&mv=m&mm
=31&pl=23&ratebypass=yes&mime=video/mp4&gir=yes&clen=43100115&lmt=14293098351419
49&dur=133.132&fexp=900720,904833,907263,916602,916631,934954,938028,9406690,940
6813,9407115,9407433,9408093,9408209,9408347,9408708,9409061,9412465,947233,9481
24,948703,951703,952612,952626,952637,957201&mt=1430089697&key=dg_yt0&sver=3&sig
nature=17B70F77DA9C13D6C56D8CA96F6EFFF8607D10AB.38A71779D74AF8BB522F6BF025B33712
A7ACBF4A&upn=ChXw6E9QUqg&ip=24.108.145.53&ipbits=0&expire=1430111433&sparams=ip,
ipbits,expire,id,itag,source,requiressl,ms,mv,mm,pl,ratebypass,mime,gir,clen,lmt
,dur'
[download] Destination: C:\BATMAN.f137.SUPERMAN
[download] 100% of 41.10MiB in 00:33
[debug] Invoking downloader on 'https://r8---sn-ni5f-t8gl.googlevideo.com/videop
layback?id=2307d49e405f759e&itag=141&source=youtube&requiressl=yes&ms=au&mv=m&mm
=31&pl=23&ratebypass=yes&mime=audio/mp4&gir=yes&clen=4247793&lmt=142930979147482
8&dur=133.212&fexp=900720,904833,907263,916602,916631,934954,938028,9406690,9406
813,9407115,9407433,9408093,9408209,9408347,9408708,9409061,9412465,947233,94812
4,948703,951703,952612,952626,952637,957201&mt=1430089697&key=dg_yt0&sver=3&sign
ature=2CA48F47F3CA6B379D5AA3F2BD78A67900F57726.21482A219B980AF0500726B1905FE42BE
B60EB65&upn=ChXw6E9QUqg&ip=24.108.145.53&ipbits=0&expire=1430111433&sparams=ip,i
pbits,expire,id,itag,source,requiressl,ms,mv,mm,pl,ratebypass,mime,gir,clen,lmt,
dur'
[download] Destination: C:\BATMAN.f141.SUPERMAN
[download] 100% of 4.05MiB in 00:01
[ffmpeg] Merging formats into "C:\BATMAN.SUPERMAN"
[debug] ffmpeg command line: ffmpeg -y -i 'C:\BATMAN.f137.SUPERMAN' -i 'C:\BATMA
N.f141.SUPERMAN' -c copy -map 0:v:0 -map 1:a:0 'C:\BATMAN.temp.SUPERMAN'
ERROR: C:\BATMAN.temp.SUPERMAN: Invalid argument
Traceback (most recent call last):
  File "youtube_dl\YoutubeDL.pyo", line 1520, in post_process
  File "youtube_dl\postprocessor\ffmpeg.pyo", line 589, in run
  File "youtube_dl\postprocessor\ffmpeg.pyo", line 148, in run_ffmpeg_multiple_f
iles
FFmpegPostProcessorError

Whereas the following command does not fail
youtube-dl -v "https://www.youtube.com/watch?v=IwfUnkBfdZ4" -o "C:\BATMAN.SUPERMAN.MP4"

**Comment by jaimeMF** [comment]
You should just end the output template with `.%(ext)s`. Although we could make it work without the correct extension.

**Comment by RingoTheDog** [comment]
I use YouTube-dl in a script that does some post processing and moves the file after downloading, so knowing the exact file name is handy...but I can code around that.

However (and sorry if this is a different issue)...today when I tried the .%(ext)s I received the following error:

C:\>youtube-dl.py -v "https://www.youtube.com/watch?v=5vxkKiy65iU"  -o C:\mytestfile.%(ext)s
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'-v', u'https://www.youtube.com/watch?v=5vxkKiy65iU', u'-o', u'C:\\mytestfile.%(ext)s']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2015.04.26
[debug] Python version 2.7.5 - Windows-7-6.1.7601-SP1
[debug] exe versions: ffmpeg N-34549-g13b7781, rtmpdump 2.4
[debug] Proxy map: {}
[youtube] 5vxkKiy65iU: Downloading webpage
[youtube] 5vxkKiy65iU: Extracting video information
[youtube] 5vxkKiy65iU: Downloading DASH manifest
WARNING: You have requested formats uncompatible for merge. The formats will be merged into mkv
[debug] Invoking downloader on 'https://r5---sn-cxaaj5o5q5-tt1d.googlevideo.com/videoplayback?id=e6fc642a2cbae625&itag=243&source=youtube&requiressl=yes&mm=31&mv=m&ms=au&pl=24&pcm2cms=yes&ratebypass=y
es&mime=video/webm&gir=yes&clen=4436785&lmt=1399016293855683&dur=117.200&mt=1430157346&sver=3&upn=ISmuEc1UW78&fexp=900720,907263,913445,934954,936022,936117,938028,9406003,9406711,9406841,9407115,9408
255,9408259,9408347,9408704,9408823,9409174,9410705,947233,948124,948703,951703,952612,952637,957201&key=dg_yt0&signature=2187B613C05400892107C27FEAC9C8F1F0C99E0F.63797C223DA89236C4501554D0B60C7195951
B75&ip=199.85.73.12&ipbits=0&expire=1430178975&sparams=ip,ipbits,expire,id,itag,source,requiressl,mm,mv,ms,pl,pcm2cms,ratebypass,mime,gir,clen,lmt,dur'
[download] Destination: C:\mytestfile.f243.webm
[download] 100% of 4.23MiB in 00:01
[debug] Invoking downloader on 'https://r5---sn-cxaaj5o5q5-tt1d.googlevideo.com/videoplayback?id=e6fc642a2cbae625&itag=141&source=youtube&requiressl=yes&mm=31&mv=m&ms=au&pl=24&pcm2cms=yes&ratebypass=y
es&mime=audio/mp4&gir=yes&clen=3742567&lmt=1401757205212462&dur=117.353&mt=1430157346&sver=3&upn=ISmuEc1UW78&fexp=900720,907263,913445,934954,936022,936117,938028,9406003,9406711,9406841,9407115,94082
55,9408259,9408347,9408704,9408823,9409174,9410705,947233,948124,948703,951703,952612,952637,957201&key=dg_yt0&signature=2A63F799794708860DD4346C922CBA95E2631D22.78437E99574A39DAAD344848F1D7B30AFFFB9F
26&ip=199.85.73.12&ipbits=0&expire=1430178975&sparams=ip,ipbits,expire,id,itag,source,requiressl,mm,mv,ms,pl,pcm2cms,ratebypass,mime,gir,clen,lmt,dur'
[download] Destination: C:\mytestfile.f141.m4a
[download] 100% of 3.57MiB in 00:01
[ffmpeg] Merging formats into "C:\mytestfile.mkv"
[debug] ffmpeg command line: ffmpeg -y -i 'C:\mytestfile.f243.webm' -i 'C:\mytestfile.f141.m4a' -c copy -map 0:v:0 -map 1:a:0 'C:\mytestfile.temp.mkv'
ERROR: C:\mytestfile.f243.webm: could not find codec parameters
Traceback (most recent call last):
  File "C:\transmogrifier\youtube-dl.py\youtube_dl\YoutubeDL.py", line 1520, in post_process
    files_to_delete, info = pp.run(info)
  File "C:\transmogrifier\youtube-dl.py\youtube_dl\postprocessor\ffmpeg.py", line 589, in run
    self.run_ffmpeg_multiple_files(info['__files_to_merge'], temp_filename, args)
  File "C:\transmogrifier\youtube-dl.py\youtube_dl\postprocessor\ffmpeg.py", line 148, in run_ffmpeg_multiple_files
    raise FFmpegPostProcessorError(msg)
FFmpegPostProcessorError




Thanks
Ringo

**Comment by dstftw** [comment]
Try updating ffmpeg.

**Comment by RingoTheDog** [comment]
Thanks dstftw.  You are correct, it was an old FFMPEG.
Just in case this helps anyone else:  While I had downloaded a new FFMPEG, I have ImageMagick installed (under windows) so the ImageMagick directory is added to the system path.  There was an older version of FFMPEG in the ImageMagick directory and so that old version was being called.  
Updated the FFMPEG in the ImageMagick directory and now "https://www.youtube.com/watch?v=5vxkKiy65iU" works correctly.
*DOH*!

(and thanks again!)
Ringo
